course_info/tests/jasmine/directives/BadgeDirectiveSpec.js

```diff
@@ -0,0 +1,45 @@
+describe('Unit testing BadgeDirective', function() {
+    beforeEach(module('CourseInfo'));
+
+    var $compile, $rootScope;
+    beforeEach(inject(function(_$compile_, _$rootScope_) {
+        $compile = _$compile_;
+        $rootScope = _$rootScope_;
+    }));
+
+    // sanity check the test framework
+    it('should inject the providers we requested', function() {
+        [$compile, $rootScope].forEach(function(thing) {
+            expect(thing).not.toBeUndefined();
+            expect(thing).not.toBeNull();
+        });
+    });
+
+    // these are just copied from BadgeDirective.js, and should be kept in sync
+    var roleToBadge = {
```

Comment:
Nah, I just err on the side of not sharing code between tests and the code under test if I can help it.  Happy to do it that way if you'd prefer.